server/neptune/workflows/internal/deploy/terraform/runner.go

```diff
@@ -78,7 +78,7 @@ func (r *WorkflowRunner) buildRequestRoot(root terraformActivities.Root, diffDir
 	var approvalType terraformActivities.PlanApprovalType
 	var reasons []string
 
-	if diffDirection == activities.DirectionDiverged || root.Trigger == terraformActivities.ManualTrigger {
+	if diffDirection == activities.DirectionDiverged || (root.Trigger == terraformActivities.ManualTrigger && !root.Rerun) {
```

Comment:
what is root here? Does this represent the branch of the project?

Reply:
root is our way of differentiating terraform config directories.

ie. production, staging are two different roots.

Reply:
We've shoved a few things in there that kinda don't make as much sense from a data modeling standpoint unfortunately.

```go
// Root is the definition of a root
type Root struct {
	Name string

	// Path is the relative path from the repo
	Path      string
	TfVersion string
	Apply     execute.Job
	Plan      PlanJob
	Trigger   Trigger
	Rerun     bool
}
```

but i guess it just contains everything you need to know about how a root is 'deployed'

Reply:
Ah got ya terraform root i see i see